### PR TITLE
Add ukernel to e2e testing

### PR DESCRIPTION
**Pull request by newling**

### CODE REVIEW

File: build_tools/ci/run_matmul_test.sh [hunk 1]
```diff
@@ -366,15 +372,40 @@ function run_matmul_test() {
...
+                      --iree-amdaie-use-pipeline=${pipeline} \
+                      --iree-amd-aie-peano-install-dir=${peano_install_path} \
+                      --iree-amd-aie-mlir-aie-install-dir=${mlir_aie_install_path} \
+                      --iree-amd-aie-vitis-install-dir=${vitis_path} \
+                      --iree-hal-dump-executable-files-to=$PWD \
+                      --iree-amd-aie-show-invoked-commands --mlir-print-ir-after-all --mlir-print-ir-module-scope --mlir-disable-threading"
+
+  if [ $use_ukernel -ne 0 ]; then
+
+    compilation_flags="${compilation_flags} --mlir-elide-elementsattrs-if-larger=2 \
```
Comment (Abhishek-Varma) [inline build_tools/ci/run_matmul_test.sh:385]: `--mlir-elide-elementsattrs-if-larger=2` - we can either get rid of this flag.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp [hunk 2]
```diff
@@ -152,7 +152,7 @@ FailureOr<ParameterSetting> ParameterSetting::create(linalg::LinalgOp linalgOp,
   const uint64_t K = lhsShape[1];
 
   FailureOr<unsigned> maybeScaleFactor =
-      getTilingScaleFactor(initType.getElementType());
+      getTilingScaleFactor(lhsType.getElementType());
```
Comment (Abhishek-Varma) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:155]: So we are to not base it on the accumulator but the input element types ? I believe this is in-line with the observations made for `INF NAN` issue ?

Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:155]: That's correct, I believe this is the change which caused `INF NAN`. This is still a very experimental PR!

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToUKernels.cpp [hunk 3]
```diff
@@ -232,8 +232,8 @@ void AMDAIELowerToUKernelsPass::runOnOperation() {
                                                             pathToUkernels);
   patterns.insert<LowerToUKernelPattern<linalg::MatmulOp>>(context, allTargets,
                                                            pathToUkernels);
-  patterns.insert<LowerToUKernelPattern<linalg::FillOp>>(context, allTargets,
-                                                         pathToUkernels);
```
Comment (newling) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToUKernels.cpp:236]: @Abhishek-Varma is this supported on mlir-aie yet? Doesn't seem to work on my branch.

**Comment by newling** [comment]
The current CI failure is a chess linking error:

```
/opt/Xilinx/Vitis/2023.1/aietools/bin/unwrapped/lnx64.o/xchesscc: error while loading shared libraries: libgurobi90.so: cannot open shared object file: No such file or directory
```


On my nuc50 machine (not CI) I can see libgurobi here:

```
jnewling@xsjipunuc50:/proj/xbuilds/2023.2_released/installs/lin64/Vitis/2023.2/aietools/lib/lnx64.o
```

on the CI machine there is no libgurobi in the Vitis directory:

```
svcnod@sharkbox1:/opt/Xilinx/Vitis$ find . -name "*urob*" -type f
svcnod@sharkbox1:/opt/Xilinx/Vitis$
```

It looks like the VITIS installed on the CI machine has got much fewer dynamic libraries installed with it. 

@spaceotter @erieaton-amd  do you know what is the history of the VITIS on the CI machine? Can it be updated to the same one on the nuc50 machine that I use?

**Comment by spaceotter** [comment]
It might have Riallto disguised as Vitis. I'm not sure I ever installed the full Vitis on sharkbox.

**Review by Abhishek-Varma — APPROVED** [review]
Thanks @newling !

**Comment by newling** [comment]
> It might have Riallto disguised as Vitis. I'm not sure I ever installed the full Vitis on sharkbox.

It turns out we don't need chess to use microkernels, so I didn't even need vitis.

**Pull request merged**